src/payment/payment.repository.ts

```diff
@@ -0,0 +1,60 @@
+import { Payment } from './entitites/payment.entity';
+
+/**
+ * Repository class for storing payments in memory.
+ * It is needed besides the queue to keep track of the payments for manual updates.
+ */
+export class PaymentRepository {
+  private payments: Payment[] = [];
+
+  /**
+   * Creates a new payment and adds it to the repository.
+   * @param payment - The payment to be created.
+   * @returns The created payment.
+   */
+  public create(payment: Payment): Payment {
```

Comment:
Are there any checks elsewhere that prevent a payment from being added multiple times?

Reply:
I think, there is no way to test for that consistently.
This repository just temporarily stores payments, that were not yet updated yet in the memory. It has no "thruth claim" since the service could have been restarted at some point.
The payment / shipment services should never send two registrations.
Maybe I could add an error log but since its no persistet repository, I would not add something more.

Reply:
Fine for me.